### PR TITLE
Don't log urlencoded access_tokens

### PR DESCRIPTION
**Pull request by erikjohnston**

### CODE REVIEW

File: synapse/app/homeserver.py [hunk 1]
```diff
@@ -496,8 +496,8 @@ def __repr__(self):
 
     def get_redacted_uri(self):
         return re.sub(
-            r'(\?.*access_token=)[^&]*(.*)$',
-            r'\1<redacted>\2',
+            r'(\?.*accesss(_|%5[Ff])token=)[^&]*(.*)$',
```
Comment (NegativeMjark) [inline synapse/app/homeserver.py]: "access" is traditionally spelt with 2 "s"s.

File: synapse/app/homeserver.py [hunk 2]
```diff
@@ -496,8 +496,8 @@ def __repr__(self):
 
     def get_redacted_uri(self):
         return re.sub(
```
Comment (NegativeMjark) [inline synapse/app/homeserver.py]: You could compile the regexp at the module level.

File: synapse/app/homeserver.py (hunk 1 above)
Comment (erikjohnston) [inline synapse/app/homeserver.py]: HUSH YOU

**Comment by erikjohnston** [comment]
PTAL

**Comment by NegativeMjark** [comment]
LGTM

**Comment by erikjohnston** [comment]
please test this

**Comment by erikjohnston** [comment]
matrixbot: test this please

**Pull request merged**